### PR TITLE
fix: add onCameraChange callback to fix android issue

### PR DESCRIPTION
**Pull request by reidzeibel**
Closes https://github.com/AtB-AS/kundevendt/issues/16093

Working on this fix with @KristianSelnas . 

Issue is caused by RNMapBox issue here : https://github.com/rnmapbox/maps/issues/2902, so basically, in Android, if you show UserLocation, then `onMapIdle` callback no longer triggers properly. In iOS this is not a problem. A workaround offered by one of the RNMapBox developer is to use a debounced `onCameraChanged` event to trigger the change, as mentioned here : https://github.com/rnmapbox/maps/issues/2902#issuecomment-1633003164.

So we are implementing that solution, but currently we have some issues where we can't find a way to implement the enforced eslint exhaustive deps on useEffect. 

Therefore hopefully someone have an idea on how to do it properly. 

This PR already fixes the issue, but the code needs to be improved while keeping the solution working.

### CODE REVIEW

**Comment by marius-at-atb** [comment]
Related issue: https://github.com/AtB-AS/mittatb-app/pull/4045

**Comment by marius-at-atb** [comment]
I'm guessing the answer is yes, but does the same Android issue occur when using `<NativeUserLocation />` instead of `UserLocation` with `renderMode={UserLocationRenderMode.Native}`?

File: src/components/map/Map.tsx [hunk 1]
```diff
@@ -73,6 +76,24 @@ export const Map = (props: MapProps) => {
...
+  useEffect(() => {
+    if (debouncedState) {
+      loadMobility({
+        visibleBounds: [
+          debouncedState.properties.bounds.ne,
+          debouncedState.properties.bounds.sw,
+        ],
+        zoomLevel: debouncedState.properties.zoom,
+        center: debouncedState.properties.center,
+      });
```
Comment (marius-at-atb) [inline src/components/map/Map.tsx:88]:
```suggestion
      onMapIdle(debouncedState)
```

File: src/components/map/Map.tsx [hunk 2]
```diff
@@ -73,6 +76,24 @@ export const Map = (props: MapProps) => {
...
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (marius-at-atb) [inline src/components/map/Map.tsx:90]: With the suggestion above, maybe it will work when wrapping `onMapIdle` and `loadMobility` with `useCallback`?

Comment (reidzeibel) [reply src/components/map/Map.tsx:90]: Unfortunately, no, it still triggers multiple debounce due to location change if we put the dependency to the useEffect.

**Comment by reidzeibel** [comment]
@marius-at-atb we are using `renderMode={UserLocationRenderMode.Native}`, can't seem to find `NativeUserLocation` on the RNMapbox, can you point out where it is?

**Comment by marius-at-atb** [comment]
https://github.com/rnmapbox/maps/blob/main/docs/NativeUserLocation.md
Might require rnmapbox to be upgraded

**Comment by reidzeibel** [comment]
@gorandalum @marius-at-atb please hold off reviewing this, I think i found something when trying to update the RNMapBox dependency version, will let you know after lunch the latest. 

@KristianSelnas

**Comment by reidzeibel** [comment]
Closing this PR because there is a better solution: Updating the RNMapBox dependency, works on both Android and iOS normally.

**Comment by toretefre** [comment]
I guess my review is not needed anymore as well. Hope dependency update works 🤞

**Pull request closed without merging**